core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala

```diff
@@ -216,6 +216,34 @@ class DynamicBrokerReconfigurationTest extends ZooKeeperTestHarness with SaslSet
     verifySslConfig("", invalidSslProperties, configDesc)
   }
 
+  @Test
+  def testKeyStoreAlter_ConfigProvider(): Unit = {
```

Comment:
We should rename this test `testUpdatesUsingConfigProvider` and keep all testing for config providers within this class. DynamicBrokerReconfigurationTest and the underlying updates are very complex and adding config providers in existing tests makes it even more difficult to debug. Having all config provider tests in one method would isolate provider related issues from other config update issues.

Reply:
I renamed the new testing function to testUpdatesUsingConfigProvider and put all testing cases to the function.

Reply:
I removed the function. Reuse function verifyState.